include/aws/mqtt/private/v5/mqtt5_service_topic_manager.h

```diff
@@ -0,0 +1,86 @@
+#ifndef AWS_MQTT5_PRIVATE_SERVICE_TOPIC_MANAGER_H
+#define AWS_MQTT5_PRIVATE_SERVICE_TOPIC_MANAGER_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/common/hash_table.h>
+#include <aws/mqtt/exports.h>
+
+struct aws_mqtt5_service_topic_node {
+
+    /* The topic name, key of the node. */
+    struct aws_byte_cursor *topic;
+
+    /* user defined data type */
+    void *userdata;
```

Comment:
The `userdata` will be a structure defined and passed by service client. The `userdata` stores the user callbacks for each topic.